packages/components/src/input-control/index.tsx

```diff
@@ -85,6 +86,8 @@ export function UnforwardedInputControl(
 				isPressEnterToChange={ isPressEnterToChange }
 				onKeyDown={ onKeyDown }
 				onValidate={ onValidate }
+				paddingInlineStart={ prefix ? space( 2 ) : undefined }
+				paddingInlineEnd={ suffix ? space( 2 ) : undefined }
```

Comment:
Instead of adding the internal `paddingInlineStart` and `paddingInlineEnd` internal props, we could also consider adding two `boolean` props called something like `reducedPaddingInlineStart` and `reducedPaddingInlineEnd`?

This is mostly to minimize exposing design-related props, and instead expose props that are more "semantic" instead (which I believe make the component more resilient to design changes).

Reply:
Exactly, I did consider booleans (called `hasPrefix`/`hasSuffix`, for example). The upside, from a maintainer's standpoint, is that we would make fewer or no promises on what effect the prop is going to have on the visual styling, especially in regards to the different size variants. The downside is that, it's less generic, so it wouldn't be flexible enough to cover possible cases where someone needs more granular control over the padding.

In any case, `InputField` itself is internal-only, so I'm not overly concerned about the specific API and contracts at this point. It's quite possible that we'll never expose custom padding props at the top-level `InputControl`. I slightly prefer to keep it flexible for now, since we're still kind of in the middle of discovering use cases for this custom padding.